synapse/lib/storm.py

```diff
@@ -488,18 +501,56 @@ def cmpr(tufo):
 
     def _cmprCtorTag(self, oper):
         tag = self._reqOperArg(oper,'valu')
-
-        props = {}
-        def cmpr(tufo):
+        glob_props = collections.defaultdict(set)
+        reg_props = {}
+
+        if tag.startswith(self._rt_glob_smmark) or self._rt_glob_smark in tag or self._rt_glob_mmark in tag:
+            # Prep the tag regex
+            _tag = tag.lower()
+            if _tag.startswith(self._rt_glob_smmark):
+                _tag = _tag.replace(self._rt_glob_smmark, self._rt_glob_smsentinal, 1)
+            while self._rt_glob_smark in _tag:
+                _tag = _tag.replace(self._rt_glob_smark, self._rt_glob_ssentinal)
+            while self._rt_glob_mmark in _tag:
+                _tag = _tag.replace(self._rt_glob_mmark, self._rt_glob_msentinal)
+            # Escape and anchor
+            _tag = re.escape(_tag)
+            _tag = _tag + '$'
+            # Substitute in regex values
+            tag_regex = _tag.replace(self._rt_glob_smsentinalbase, self._rt_glob_mre)
+            tag_regex = tag_regex.replace(self._rt_glob_ssentinalbase, self._rt_glob_sre)
+            tag_regex = tag_regex.replace(self._rt_glob_msentinalbase, self._rt_glob_mre)
+
+        def glob_cmpr(tufo):
+            form = tufo[1].get('tufo:form')
+            # Check cached props first
+            fprops = glob_props.get(form, ())
+            for prop in fprops:
+                if prop in tufo[1]:
+                    return True
+            # Now search for matching tags on the tufo
+            form_prefix = '*|%s|' % form
+            full_regex = re.escape(form_prefix) + tag_regex
```

Comment:
fyi - The actual re.search/match operation is cached at the module level (of re) in a local cache.
https://github.com/python/cpython/blob/master/Lib/re.py#L268

Will explore optimizations here.